jetty-util/src/main/java/org/eclipse/jetty/util/compression/CompressionPool.java

```diff
@@ -56,73 +65,102 @@ public CompressionPool(int capacity)
     /**
      * @return Object taken from the pool if it is not empty or a newly created Object
      */
-    public T acquire()
+    public Entry acquire()
     {
-        T object;
-
-        if (_capacity == 0)
-            object = newObject();
-        else
+        Entry entry = null;
+        if (_pool != null)
         {
-            object = _pool.poll();
-            if (object == null)
-                object = newObject();
-            else if (_capacity > 0)
-                _numObjects.decrementAndGet();
+            Pool<Entry>.Entry acquiredEntry = _pool.acquire(e -> new Entry(newObject()));
+            if (acquiredEntry != null)
+            {
+                entry = acquiredEntry.getPooled();
+                entry.setEntry(acquiredEntry);
+            }
```

Comment:
It is confusing to have your Entry called Entry.... but eitherway, don't set the entry on the entry, instead pass it into the constructor and make it final:
```suggestion
            Pool<Entry>.Entry acquiredEntry = _pool.acquire(e -> new Entry(newObject(), e));
            if (acquiredEntry == null)
                return new Entry(newObject(), null);
            return acquiredEntry.getPooled();
```

Reply:
can we rename `newObject` to `newCompressor`

Reply:
Its not really `newCompressor`, an inflater is the opposite of a compressor.
But this is only internal implementation methods, they should not be available publicly.

Reply:
`newPooled` ?